src/libsyntax/ast.rs

```diff
@@ -399,6 +399,16 @@ impl Generics {
     pub fn is_parameterized(&self) -> bool {
         self.is_lt_parameterized() || self.is_type_parameterized()
     }
+    pub fn span_for_name(&self, name: &str) -> Option<Span> {
+        let mut span = None;
+        for t in &self.ty_params {
+            if &format!("{}", t.ident.name.as_str()) == name {
+                span = Some(t.span);
+                break;
```

Comment:
You can `return Some(t.span);` instead of breaking, and get rid of the local variable `span`.